sdk/messaging/azeventhubs/mgmt.go

```diff
@@ -0,0 +1,201 @@
+	token, err := ns.GetTokenForEntity(eventHub)
+
+	if err != nil {
+		return EventHubProperties{}, internal.TransformError(err)
+	}
+
+	amqpMsg := &amqp.Message{
+		ApplicationProperties: map[string]any{
+			"operation":      "READ",
+			"name":           eventHub,
+			"type":           "com.microsoft:eventhub",
+			"security_token": token.Token,
+		},
+	}
+
+	resp, err := rpcLink.RPC(context.Background(), amqpMsg)
+
+	if err != nil {
+		return EventHubProperties{}, err
+	}
+
+	if resp.Code >= 300 {
+		return EventHubProperties{}, fmt.Errorf("failed getting partition properties: %v", resp.Description)
+	}
+
+	return newEventHubProperties(resp.Message.Value)
+}
+
+type PartitionProperties struct {
+	BeginningSequenceNumber    int64
+	EventHubName               string
+	IsEmpty                    bool
+	LastEnqueuedOffset         int64
+	LastEnqueuedOn             time.Time
+	LastEnqueuedSequenceNumber int64
+	PartitionId                string
+}
+
+type GetPartitionPropertiesOptions struct {
+	// For future expansion
+}
+
+// getPartitionProperties gets properties for a specific partition. This includes data like the last enqueued sequence number, the first sequence
+// number and when an event was last enqueued to the partition.
+func getPartitionProperties(ctx context.Context, ns internal.NamespaceForManagementOps, rpcLink internal.RPCLink, eventHub string, partitionID string, options *GetPartitionPropertiesOptions) (PartitionProperties, error) {
+	token, err := ns.GetTokenForEntity(eventHub)
+
+	if err != nil {
+		return PartitionProperties{}, err
+	}
+
+	amqpMsg := &amqp.Message{
+		ApplicationProperties: map[string]any{
+			"operation":      "READ",
+			"name":           eventHub,
+			"type":           "com.microsoft:partition",
+			"partition":      partitionID,
+			"security_token": token.Token,
+		},
+	}
+
+	resp, err := rpcLink.RPC(context.Background(), amqpMsg)
+
+	if err != nil {
+		return PartitionProperties{}, internal.TransformError(err)
+	}
+
+	if resp.Code >= 300 {
+		return PartitionProperties{}, fmt.Errorf("failed getting partition properties: %v", resp.Description)
+	}
+
+	return newPartitionProperties(resp.Message.Value)
+}
+
+func newEventHubProperties(amqpValue any) (EventHubProperties, error) {
+	m, ok := amqpValue.(map[string]any)
+
+	if !ok {
+		return EventHubProperties{}, nil
+	}
+
+	partitionIDs, ok := m["partition_ids"].([]string)
+
+	if !ok {
+		return EventHubProperties{}, fmt.Errorf("invalid message format")
+	}
+
+	name, ok := m["name"].(string)
+
+	if !ok {
+		return EventHubProperties{}, fmt.Errorf("invalid message format")
+	}
+
+	createdOn, ok := m["created_at"].(time.Time)
+
+	if !ok {
+		return EventHubProperties{}, fmt.Errorf("invalid message format")
+	}
+
+	return EventHubProperties{
+		Name:         name,
+		CreatedOn:    createdOn,
+		PartitionIDs: partitionIDs,
+	}, nil
+}
+
+func newPartitionProperties(amqpValue any) (PartitionProperties, error) {
```

Comment:
Why is the type `any` if the only acceptable type is `map[string]any`?

Reply:
This is actually the amqpMessage.Value field, so it's typed 'any' already (more of the 'dynamic' stuff)